src/js/tools/metroconfig.ts

```diff
@@ -146,6 +158,29 @@ function withSentryDebugId(config: MetroConfig): MetroConfig {
   };
 }
 
+/**
+ *
+ */
+export function excludeSentryWebReplay(config: MetroConfig, includeWebReplay: boolean | undefined): MetroConfig {
+  const originalResolver = config.resolver?.resolveRequest;
+
+  return {
+    ...config,
+    resolver: {
+      ...config.resolver,
+      resolveRequest: (context, moduleName, platform) => {
+        if (includeWebReplay === false || (platform !== 'web' && moduleName.includes('@sentry/replay'))) {
```

Comment:
After the flag change this will simplify to

```suggestion
        if (excludeWebReplay && moduleName.includes('@sentry/replay')) {
```